apps/server/src/modules/learnroom/service/course.service.ts

```diff
@@ -68,4 +68,8 @@ export class CourseService {
 	private getCoursesId(courses: Course[]): EntityId[] {
 		return courses.map((course) => course.id);
 	}
+
+	async findOneForUser(courseId: EntityId, userId: EntityId): Promise<Course> {
+		return this.repo.findOne(courseId, userId);
```

Comment:
we should avoid returning logic
const coursePromise = this.repo.findOne(courseId, userId);
return coursePromise;